protected.go

```diff
@@ -151,39 +182,26 @@ func (p *Protector) dial(op ops.Op, network, addr string, timeout time.Duration)
 	default:
 		return nil, errors.New("Unsupported network: %v", network)
 	}
-
-	start := time.Now()
-
-	remainingTimeout := func() time.Duration {
-		return timeout - time.Now().Sub(start)
-	}
-
 	host, port, err := SplitHostPort(addr)
 	if err != nil {
 		return nil, err
 	}
 
-	_ip, timedOut, err := withtimeout.Do(remainingTimeout(), func() (interface{}, error) {
-		// do DNS query
-		ip := net.ParseIP(host)
-		if ip != nil {
-			return ip, nil
-		}
+	ip := net.ParseIP(host)
+	if ip == nil {
 		// Try to resolve it
 		addr, err := p.Resolve(network, addr)
```

Comment:
The call to `p.protect` and `conn. connectSocket` are not context-aware, so this can be blocked anyway.